pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -580,18 +581,22 @@ def __resolve_ckpt_dir(self, trainer: "pl.Trainer") -> None:
         if self.dirpath is not None:
             return  # short circuit
 
-        if trainer.logger is not None:
+        if trainer.loggers:
             if trainer.weights_save_path != trainer.default_root_dir:
                 # the user has changed weights_save_path, it overrides anything
                 save_dir = trainer.weights_save_path
             else:
-                save_dir = trainer.logger.save_dir or trainer.default_root_dir
-
+                if len(trainer.loggers) == 1 and trainer.loggers[0].save_dir:
+                    save_dir = trainer.loggers[0].save_dir
+                else:
+                    save_dir = trainer.default_root_dir
+            # TODO: Find out we handle trainer.logger.version without LoggerCollection
```

Comment:
How should we handle trainer.logger.version and trainer.logger.name here? LoggerCollection will concatenate the values together from each logger, but with trainer.loggers we no longer have that logic and that logic might not be the best approach anyways.